scripts/metric_reporter/junit_xml_parser.py

```diff
@@ -0,0 +1,186 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Module for parsing test suite results from JUnit XML content."""
+
+import logging
+import re
+from pathlib import Path
+from typing import Any
+
+import defusedxml.ElementTree as ElementTree
+from pydantic import BaseModel, ValidationError
+
+from scripts.common.error import BaseError
+
+
+class Property(BaseModel):
+    """Represents a property of a test case."""
+
+    name: str
+    value: str
+
+
+class Skipped(BaseModel):
+    """Represents a skipped test case."""
+
+    reason: str | None = None
+
+
+class Failure(BaseModel):
+    """Represents a failure of a test case."""
+
+    message: str
+    type: str | None = None
+    text: str | None = None
+
+
+class JUnitXMLSystemOut(BaseModel):
+    """Represents system out information."""
+
+    text: str | None = None
+
+
+class JUnitXMLTestCase(BaseModel):
+    """Represents a test case in a test suite."""
+
+    name: str
+    classname: str | None = None
+    time: float | None = None
+    properties: list[Property] | None = None
+    skipped: Skipped | None = None
+    failure: Failure | None = None
+    system_out: JUnitXMLSystemOut | None = None
```

Comment:
We encountered `system_err` in Playwright at the beginning of the year, but haven't since and there is a [guard against it](https://github.com/mozilla/fxa/blob/main/packages/functional-tests/.eslintrc.js#L17) in FxA now I believe.

I have opted not to include it letting the error on L121 be emitted. My thinking is that because we know that the presence of this tag is problematic for CI, if we should encounter it, I want to be properly notified so that we can intervene. Since we are still running scripts and this isn't a service yet I felt this was okay.